src/helper.cc

```diff
@@ -681,12 +681,12 @@ statefulhelper::submit(const char *buf, HLPCB * callback, void *data, const Help
         helper_stateful_server *srv;
         if ((srv = StatefulGetFirstAvailable(this))) {
             reserveServer(srv);
-            helperStatefulDispatch(srv, r);
+            helperStatefulDispatch(srv, r); // may delete r
         } else
             StatefulEnqueue(this, r);
     }
 
-    debugs(84, DBG_DATA, "placeholder: '" << r->request.placeholder <<
+    debugs(84, DBG_DATA, "placeholder: '" << (r ? r->request.placeholder: -1) <<
            "', " << Raw("buf", buf, (!buf?0:strlen(buf))));
 
```

Comment:
The bug looks real, but this code change does not match the new comment or the PR purpose: Deleting `r` does not make `r` nil or falsy.

```suggestion
    // r may be dangling here
```

... or set `r` to nil after helperStatefulDispatch() and, just in case, StatefulEnqueue() calls above.

Reply:
I do not see the point of having this logged so late in the process. Surely something should be logged on `Helper::Xaction` construct, then maybe on whatever action is taken (not here).

Reply:
Okay, so it seems that there is consensus that this logging statement is not useful,
which is a different problem than the one I was aiming to solve (r being possibly dangling).

@rousskov 's solution works for that, so I'll bring it in and be happy :)